components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/IdentityUserMetadataMgtHandler.java

```diff
@@ -0,0 +1,133 @@
+            handlePostAuthenticate(eventProperties, userStoreManager);
+        } else if (IdentityEventConstants.Event.POST_UPDATE_CREDENTIAL.equals(event.getEventName()) ||
+                IdentityEventConstants.Event.POST_UPDATE_CREDENTIAL_BY_ADMIN.equals(event.getEventName())) {
+            handleCredentialUpdate(eventProperties, userStoreManager);
+        }
+    }
+
+    private void handlePostAuthenticate(Map<String, Object> eventProperties, UserStoreManager userStoreManager)
+            throws IdentityEventException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Start handling post authentication event.");
+        }
+        if ((Boolean) eventProperties.get(IdentityEventConstants.EventProperty.OPERATION_STATUS)) {
+            String lastLoginTime = Long.toString(System.currentTimeMillis());
+            setUserClaim(userStoreManager, eventProperties, IdentityMgtConstants.LAST_LOGIN_TIME,
+                    lastLoginTime, POST_AUTHENTICATION);
+        }
+    }
+
+    private void handleCredentialUpdate(Map<String, Object> eventProperties, UserStoreManager userStoreManager)
+            throws IdentityEventException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Start handling post credential update event.");
+        }
+        String lastPasswordUpdateTime = Long.toString(System.currentTimeMillis());
+        setUserClaim(userStoreManager, eventProperties, IdentityMgtConstants.LAST_PASSWORD_UPDATE_TIME,
+                lastPasswordUpdateTime, POST_CREDENTIAL_UPDATE);
+    }
+
+    private void setUserClaim(UserStoreManager userStoreManager, Map<String, Object> eventProperties,
+                              String claimURI, String claimValue, String eventName) throws IdentityEventException {
+
+        String username = (String) eventProperties.get(IdentityEventConstants.EventProperty.USER_NAME);
+        Map<String, String> userClaims = new HashMap<>();
+        userClaims.put(claimURI, claimValue);
+        try {
+            userStoreManager.setUserClaimValues(username, userClaims, null);
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("Successfully updated the user claims related to %s event.", eventName));
+            }
+        } catch (UserStoreException e) {
+            throw new IdentityEventException(
+                    String.format("Error occurred while updating user claims related to %s event.", eventName), e);
+        }
+    }
+
+    @Override
+    public String getName() {
+
+        return "identityUserMetadataMgtHandler";
+    }
+
+    @Override
+    public int getPriority(MessageContext messageContext) {
+
+        return 50;
+    }
+
+    @Override
+    public void init(InitConfig configuration) throws IdentityRuntimeException {
+
+        super.init(configuration);
+    }
+
+    public String getFriendlyName() {
+
+        return "Identity User Metadata Management Handler";
```

Comment:
Shall we replace "Identity User Metadata Management Handler" with a constant? WDYT?